imap_processing/ena_maps/ena_maps.py

```diff
@@ -0,0 +1,543 @@
+        Must contain the Spice frame in which the pixel centers are defined.
+    spatial_object_output_frame : PointingSet | AbstractSkyMap
+        The object containing a grid or tessellation of spatial pixels
+        into which the input spatial pixel centers will 'land', and be matched to
+        corresponding pixel 1D indices in the output frame.
+    event_time : float, optional
+        Event time at which to transform the input spatial object to the output frame.
+        This can be manually specified, e.g., for converting between Maps which do not
+        contain an epoch value.
+        The default value is None, in which case the event time of the PointingSet
+        object is used.
+
+    Returns
+    -------
+    flat_indices_input_grid_output_frame : NDArray
+        1D array of pixel indices of the output object corresponding to each pixel in
+        the input object. The length of the array is equal to the number of pixels in
+        the input object, and may contain 0 or >1 occurrences of the same output index.
+
+    Raises
+    ------
+    ValueError
+        If both input and output objects are PointingSet objects.
+    ValueError
+        If the event time is not specified and both objects are SkyMaps.
+    NotImplementedError
+        If the output tiling type is HEALPIX. Will be implemented in the future.
+    ValueError
+        If the tiling type of the output frame is not RECTANGULAR or HEALPIX.
+    """
+    if isinstance(spatial_object_input_frame, PointingSet) and isinstance(
+        spatial_object_output_frame, PointingSet
+    ):
+        raise ValueError("Cannot match indices between two PointingSet objects.")
+
+    # If event_time is not specified, use event_time of the PointingSet, if present.
+    if event_time is None:
+        if isinstance(spatial_object_input_frame, PointingSet):
+            event_time = spatial_object_input_frame.data["epoch"].values
+        elif isinstance(spatial_object_output_frame, PointingSet):
+            event_time = spatial_object_output_frame.data["epoch"].values
+        else:
+            raise ValueError(
+                "Event time must be specified if both objects are SkyMaps."
+            )
+
+    obj1_az_el_points_frame1 = spatial_object_input_frame.az_el_points
+
+    # If the two objects are not already in the same frame,
+    # transform the input pixel centers to the output frame.
+    if (
+        spatial_object_input_frame.spice_reference_frame
+        is not spatial_object_output_frame.spice_reference_frame
+    ):
+        obj1_az_el_points_frame2 = geometry.frame_transform_az_el(
+            et=event_time,
+            az_el=obj1_az_el_points_frame1,
+            from_frame=spatial_object_input_frame.spice_reference_frame,
+            to_frame=spatial_object_output_frame.spice_reference_frame,
+            degrees=False,
+        )
+    else:
+        obj1_az_el_points_frame2 = obj1_az_el_points_frame1
```

Comment:
I am finding the original naming of `input` and `output` followed by this naming of `obj1`, `frame1`, `obj2`, `frame2` a little confusing. I wonder if this could be simplified by something like:
```suggestion
    transformed_input_az_el = spatial_object_input_frame.az_el_points
    if (
        spatial_object_input_frame.spice_reference_frame
        is not spatial_object_output_frame.spice_reference_frame
    ):
        transformed_input_az_el = geometry.frame_transform_az_el(
            et=event_time,
            az_el=obj1_az_el_points_frame1,
            from_frame=spatial_object_input_frame.spice_reference_frame,
            to_frame=spatial_object_output_frame.spice_reference_frame,
            degrees=False,
        )
```

Reply:
Another naming suggestion would be to potentially shorten the input argument to `input_object` / `output_object` and remove the spatial and frame since those are implied by using this function.

Reply:
Done! I switched to `"input_object"` and am now using the explicit formulation "`input_obj_az_el_output_frame`". It's a bit wordier but describes exactly what it is.